src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -3190,5 +3060,102 @@ subroutine convert_cvpnx_to_nx(qnx_arr,cvpnr,cvpnr_pvalue,cloud_nt_updt,q_arr,qr
     qnx_arr=tmparr_qnr
 
 end subroutine convert_cvpnx_to_nx
+   subroutine gsi_copy_bundle(bundi,bundo) 
+   use gsi_bundlemod, only:gsi_bundleinquire, gsi_bundlegetpointer,gsi_bundleputvar
+!ct   use gsi_bundlemod, only : gsi_bundle
+   implicit none  
+    
+!  copy the variables in the gsi_metguess_bundle_inout to gsi_bundle_inout or
+!  vice versa, according to icopy_flag  
+! !INPUT PARAMETERS:
+
+   type(gsi_bundle), intent(in   ) :: bundi
+
+! !INPUT/OUTPUT PARAMETERS:
+
+   type(gsi_bundle), intent(inout) :: bundo
+   character(len=max_varname_length),dimension(:),allocatable:: src_name_vars2d
+   character(len=max_varname_length),dimension(:),allocatable:: src_name_vars3d
+   character(len=max_varname_length),dimension(:),allocatable:: target_name_vars2d
+   character(len=max_varname_length),dimension(:),allocatable:: target_name_vars3d
+   character(len=max_varname_length) ::varname 
+   real(r_kind),dimension(:,:,:),pointer:: pvar3d
+   real(r_kind),dimension(:,:,:),pointer:: pvar2d
```

Comment:
While porting the GSI to WCOSS2, not nulling pointers leads to segfaults.  Please add =>NULL() to pointer declarations.